docs/concepts/artifact.mdx

```diff
@@ -0,0 +1,239 @@
+  "artifactType": "application/vnd.example+type",
+  "config": {
+    "mediaType": "application/vnd.oci.empty.v1+json",
+    "digest": "sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a",
+    "size": 2
+  },
+  "layers": [
+    {
+      "mediaType": "application/vnd.oci.empty.v1+json",
+      "digest": "sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a",
+      "size": 2
+    }
+  ],
+  "annotations": {
+    "oci.opencontainers.image.created": "2023-01-02T03:04:05Z",
+    "com.example.info": "useful-info"
+  }
+}
+```
+
+### Harnessing Image Indexes
+
+One of the key advantages of OCI artifacts is their ability to utilize [Image Indexes][image-index],
+allowing the bundling of various OCI artifacts. An example use case if is when you want to represent
+multiple SBOMs of different architectures under one unified artifact.
+
+The Image Index is a higher-level construct in the OCI Image Specification that can point to
+multiple Image Manifests, each suitable for a different platform or architecture. It helps to
+group related artifacts together, providing a single entry-point for accessing any specific
+artifact depending on the required architecture.
+
+Below is an example of an OCI Image Index for an artifact that represents multiple SBOMs
+of different architectures:
+
+```json 
+{
+  "schemaVersion": 2,
+  "artifactType": "application/vnd.example.sbom.v1",
+  "manifests": [
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "artifactType": "application/vnd.example.sbom.v1",
+      "size": 7143,
+      "digest": "sha256:1111...",
+      "platform": {
+        "architecture": "amd64",
+        "os": "linux"
+      },
+    },
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "artifactType": "application/vnd.example.sbom.v1",
+      "size": 7682,
+      "digest": "sha256:2222...",
+      "platform": {
+        "architecture": "arm64",
+        "os": "linux"
+      },
+    }
+  ]
+}
+```
+
+## Best Practices and Limitations
+
+While working with OCI artifacts, keep in mind that not all registries have implemented
+image-spec v1.1. For backward compatibility, OCI artifacts can use the `config.mediaType`
+to represent the artifact type, aligning with the IANA mediaType of the artifact.
+
+In cases where the creation of an artifact doesn't involve a config blob, make sure to use
+the `artifactType` field in the manifest. If a config blob is indispensable, comply with
+the image specification and employ an empty one. As a rule of thumb, keep the manifest
+size manageable by avoiding excessive annotations and leveraging blobs for larger data chunks.
+
+## The Road Ahead
+
+The evolution of OCI artifacts is a journey in progress. While the current specification
+does not permit an empty config blob, it's plausible that future updates might
+make layers and blobs optional—bringing more flexibility to the table.
```

Comment:
The current empty design provides the flexibility. It's just that it's hacky, which I realize we'd like to avoid saying. Suggesting the following to use "optimizing" or "working around"
> make layers and blobs optional—bringing more flexibility to the table.
make layers and blobs optional, optimizing persistence over referencing an `empty` blob.